vyper/venom/analysis/mem_ssa.py

```diff
@@ -0,0 +1,388 @@
+        # Initialize entry block with liveOnEntry
+        entry_block = self.dom.entry_block
+        self.current_def[entry_block] = self.live_on_entry
+
+        for bb in self.cfg.dfs_pre_walk:
+            self._process_block_definitions(bb)
+
+        # Second pass: insert phi nodes where needed
+        self._insert_phi_nodes()
+
+        # Third pass: connect uses to their reaching definitions
+        self._connect_uses_to_defs()
+
+    def _process_block_definitions(self, block: IRBasicBlock):
+        """Process memory definitions and uses in a basic block"""
+        effect_type = Effects.STORAGE if self.location_type == "storage" else Effects.MEMORY
+        for inst in block.instructions:
+            # Check for memory reads
+            if effect_type in inst.get_read_effects():
+                mem_use = MemoryUse(self.next_id, inst)
+                self.next_id += 1
+                self.memory_uses.setdefault(block, []).append(mem_use)
+                self.inst_to_use[inst] = mem_use
+
+            # Check for memory writes
+            if effect_type in inst.get_write_effects():
+                mem_def = MemoryDef(self.next_id, inst)
+                self.next_id += 1
+
+                mem_def.reaching_def = self._get_reaching_def_for_def(block, mem_def)
+
+                self.memory_defs.setdefault(block, []).append(mem_def)
+                self.current_def[block] = mem_def
+                self.inst_to_def[inst] = mem_def
+
+    def _insert_phi_nodes(self):
+        """Insert phi nodes at appropriate points in the CFG"""
+        worklist = list(self.memory_defs.keys())
+
+        while worklist:
+            block = worklist.pop()
+            for frontier in self.dom.dominator_frontiers[block]:
+                if frontier not in self.memory_phis:
+                    phi = MemoryPhi(self.next_id, frontier)
+                    # Add operands from each predecessor block
+                    for pred in frontier.cfg_in:
+                        reaching_def = self._get_in_def(pred)
+                        if reaching_def:
+                            phi.operands.append((reaching_def, pred))
+                    self.next_id += 1
+                    self.memory_phis[frontier] = phi
+                    worklist.append(frontier)
+
+    def _connect_uses_to_defs(self):
+        """Connect memory uses to their reaching definitions"""
+
+        for bb in self.cfg.dfs_pre_walk:
+            if bb in self.memory_uses:
+                uses = self.memory_uses[bb]
+                for use in uses:
+                    use.reaching_def = self._get_reaching_def(bb, use)
+
+    def _get_in_def(self, bb: IRBasicBlock) -> Optional[MemoryAccess]:
+        """Get the cfg in memorydefinition for a block"""
+        if bb in self.memory_phis:
+            return self.memory_phis[bb]
+
+        if bb in self.memory_defs and self.memory_defs[bb]:
+            return self.memory_defs[bb][-1]
+
+        if bb.cfg_in:
```

Comment:
style - don't use truthy:
```suggestion
        if len(bb.cfg_in) > 0:
```